Formula/xcdiff.rb

```diff
@@ -0,0 +1,20 @@
+class Xcdiff < Formula
+  desc "Tool to diff xcodeproj files"
+  homepage "https://github.com/bloomberg/xcdiff"
+  url "https://github.com/bloomberg/xcdiff.git",
+    tag:      "0.9.0",
+    revision: "0e6a933af7cada626fb3aac2fe44c5dc8fb745c6"
+  license "Apache-2.0"
+  head "https://github.com/bloomberg/xcdiff.git", branch: "main"
+
+  def install
+    system "make", "update_version"
+    system "make", "update_hash"
+    system "make", "build"
+    bin.install ".build/release/xcdiff"
+  end
+
+  test do
+    assert_match version.to_s, shell_output("#{bin}/xcdiff --version").chomp
```

Comment:
We need a test that exercises the some of the functionality of the app. Version checks or usage checks (foo --version or foo --help) are not sufficient, as explained in the formula cookbook.

In most cases, a good test would involve running a simple test case: run #{bin}/foo input.txt.

- Then you can check that the output is as expected (with assert_equal or assert_match on the output of shell_output)
- You can also check that an output file was created, if that is expected: assert_predicate testpath/"output.txt", :exist?

Some advice for specific cases:

- If the formula is a library, compile and run some simple code that links against it. It could be taken from upstream's documentation / source examples.
- If the formula is for a GUI program, try to find some function that runs as command-line only, like a format conversion, reading or displaying a config file, etc.
- If the software cannot function without credentials, a test could be to try to connect with invalid credentials (or without credentials) and confirm that it fails as expected.
- Same if the software requires a virtual machine, docker instance, etc. to be running.

Reply:
Hi @SMillerDev , many thanks for pointing it out. I was about to ask about the test. Xcdiff tests xcodeprojects and sadly those files are very long and verbose.
Does the test have access to the sources? In such case we could just use one of the existing fixtures. Alternatively we can download the fixture (not sure if that's acceptable).

Reply:
Downloading the fixture sounds good. You'll need a resource block and you'll need to name it something like `homebrew-testdata` so our automation knows it's for internal use only.

Reply:
Many thanks @SMillerDev, I will do it today.